### ISSUE TITLE
question about Current recall

### ISSUE THREAD

**Issue by ShwanMario** [experiencor/basic-yolo-keras#117]
@experiencor 

Hello,I have some problem while implement the raccoon model.
When I start warm up training,I found that my current recall increase from about 0.83 to 0.99 in the first few iterate,and then drop to zero after 1 epoch,but the loss was decrease continually.

I set warm up epoch as 3,and after the warm up,I try to predict some pictures,and none of boxes were found all the time.

I changed some settings like images' path,annotations' path and logs' path :

I have changed the path in annotation to        <path>./raccoon_dataset/images/raccoon-1.jpg</path>
which is the relative path to the image,and my annotation file is shown in the following picture.

![pic](https://user-images.githubusercontent.com/34535872/34648089-fe0dad5a-f3cd-11e7-9041-c7c083040912.PNG)

and my config file was like this:

![pic2](https://user-images.githubusercontent.com/34535872/34648088-f6f20674-f3cd-11e7-9729-3cd800dd7c08.PNG)

And then I use 
> python train.py -c config.json

to start training.But I encounter the problem mentioned above.
btw,I use python3.6,but I manually edit the print(),is this the problem?
Could u help me to solve the problem,have I did something wrong during training?

Thanks a lot!

**Comment by experiencor** [comment]
Hi @ShwanMario! This behavior is expected. At the end of the warmup, the network learns to make predictions near to the anchors. It does badly at predicting actual objects. You need to set the warmup_epochs to 0 after the warmup to actually train the network.

Just curious, why don't you go ahead with the actual networking training but stop at the warmup?

**Comment by ShwanMario** [comment]
well,thanks for reply soon!

After warm up,I set the warmup_epochs to 0 and go ahead with the actual training,but in the first 
 few iterate during epoch 1 of actual training, the current recall&average recall have been 0 , then I guess there is something wrong with the model,so I stop the actual training.

So u mean that the recall is close to 1 in the first few iterate and then drop sharply is expected,
and that I should continue the training for more epochs,and the recall was expected to increase after some epochs,right?

But I wanna know why the recall is close to 1 in the first and then drop sharply,could u please tell me why?

Thanks for your attention!

**Comment by experiencor** [comment]
@ShwanMario What I mean is that at the end of the warmup stage the recall will approach 0 as the loss function of the network is not optimized to predict object during this stage. It is optimized to make predictions near to the anchors' sizes. In the beginning of the warmup stage, the recall can be any value due to random initialization. It just happens to be close to 1 initially sometimes.

When you start the actual training, the recall will increase.

**Comment by ShwanMario** [comment]
@experiencor 
sorry for bothering u again,but I encountered an new problem.

After warm_up,I started actual training,but it stopped at 6th epoch , and the model didn't perform well when predicting raccoons.

How should I deal with this problem?

Thanks a lot!

**Comment by experiencor** [comment]
@ShwanMario Current recall of 0.6 or 0.7 is pretty good in general. It means that the predicted bounding boxes overlap with 60% the true bounding boxes (IOU of 0.6). The best model in my experience is FullYolo. You may try that. Over than that, you need more data to improve the performance.

**Comment by ShwanMario** [comment]
@experiencor 

Another question , if  I load the pre-trained weight ''full_yolo_backend.h5'' and then use it to train the model, is the model update weights for the whole network or only update weights of the last layer?

**Comment by experiencor** [comment]
@ShwanMario It will update the whole network, not just the last layer.

You can iterate through the layers and set the trainable field of a layer to false if you want to freeze that layer (https://keras.io/getting-started/faq/#how-can-i-freeze-keras-layers).

**Comment by GengCorn** [comment]
Hi @experiencor,
I tried implementing "Tiny Yolo" for the Raccoon Dataset.
I have three observations:
1) The average loss is the same (which is around 10 for the first 3 warmup epoch) if i load the pre-trained weights "tiny_yolo_racccoon" or if i started training from scratch without pre-trained weights.
2) For both cases mentioned above, the average recall plummeted to 0 after 3 epoch. And it does not seem to improve.
3) I only managed to train to 7 epochs as it runs to nump exp overflow in the sigmoid function.
I don't know what i got wrong
I am extremely grateful if you can help me out
Thank you
